projects/plutus/src/x86/ltc_adc.c

```diff
@@ -0,0 +1,91 @@
+#include "ltc_adc.h"
+
+// x86 module for mocking current sense data
+
+#include "critical_section.h"
+#include "ltc2484.h"
+
+static int32_t *s_test_voltage = NULL;
+
+static void prv_ltc_adc_read(SoftTimerID timer_id, void *context) {
+  LtcAdcStorage *storage = (LtcAdcStorage *)context;
+
+  if (s_test_voltage == NULL) {
+    storage->buffer.value = 0;
+  } else {
+    storage->buffer.value = *s_test_voltage;
+  }
+
+  if (storage->callback != NULL) {
+    storage->callback(&storage->buffer.value, storage->context);
+  }
+
+  soft_timer_start_millis(LTC2484_MAX_CONVERSION_TIME_MS, prv_ltc_adc_read, storage,
+                          &storage->buffer.timer_id);
+}
+
+StatusCode ltc_adc_init(LtcAdcStorage *storage, const LtcAdcSettings *settings) {
+  if (storage == NULL || settings->filter_mode >= NUM_LTC_ADC_FILTER_MODES) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+
+  // Since the context pointer isn't used in the x86 tests, use it to pass in a test voltage
+  if (storage->context != NULL) {
+    s_test_voltage = ((int32_t *)storage->context);
+  }
+
+  return soft_timer_start_millis(LTC2484_MAX_CONVERSION_TIME_MS, prv_ltc_adc_read, storage,
+                                 &storage->buffer.timer_id);
+}
+
+StatusCode ltc2484_raw_adc_to_uv(uint8_t *spi_data, int32_t *voltage) {
```

Comment:
Please move this into its own module instead of copy and pasting it here.

Reply:
Done

Reply:
No, you deleted it entirely. Make it its own module.